src/gam.py

```diff
@@ -503,6 +503,34 @@ def getCharSet(i):
     return (i, GC_Values.get(GC_CHARSET, GM_Globals[GM_SYS_ENCODING]))
   return (i+2, sys.argv[i+1])
 
+def cleanCourseId(courseId):
+  if courseId.startswith(u'd:'):
+    return courseId[2:]
+  return courseId
+
+def normalizeCourseId(courseId):
+  if not courseId.isdigit() and courseId[:2] != u'd:':
+    return u'd:{0}'.format(courseId)
+  return courseId
+
+def getCourseId(i):
+  if i < len(sys.argv):
+    courseId = sys.argv[i]
+    if courseId:
+      return normalizeCourseId(courseId)
+  print u'ERROR: expected a <CourseId>'
+  sys.exit(2)
+
+def getCourseAlias(i):
+  if i < len(sys.argv):
+    courseAlias = sys.argv[i]
+    if courseAlias:
+      if courseAlias[:2] != u'd:':
+        return u'd:{0}'.format(courseAlias)
+      return courseAlias
+  print u'ERROR: expected a <CourseAlias>'
+  sys.exit(2)
+
```

Comment:
These functions seem overly complicated. We should be looking to solve the minimal issue (course aliases that are numeric without adding a bunch of overhead functions.